hw27.4

Mykhaylo12 opened the pull request.

Mykhaylo12 committed:
hw27.4

Sofasmile commented on src/main/java/com/dev/cinema/dao/ShoppingCartDao.java:
```diff
@@ -9,4 +9,6 @@
 
     void update(ShoppingCart shoppingCart);
+
+    void clear(ShoppingCart shoppingCart);
```
давай не будемо використовувати такий метод на рівні дао, ми ж можемо скористатись методом апдейт. А на рівні сервісу прописати наступне:
shoppingCart.setTickets(new ArrayList<Tiket>());
        shoppingCartDao.update(shoppingCart);

Mykhaylo12 replied:
ok. A метод clear видалити?

Sofasmile replied:
на рівні сервісу залишити, а на рівні дао видалити

Mykhaylo12 replied:
ok

Sofasmile commented on src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java:
```diff
@@ -0,0 +1,40 @@
+    public List<Order> getOrderHistory(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("FROM orders WHERE user_id=:user_id", Order.class)
+                    .setParameter("user_id", user.getId()).getResultList();
```
ми тут дістаємо інформацію про тавари?

Mykhaylo12 replied:
ми тут дістаємо всі  ордери юзера. А що мені потрібно  поправити?

Sofasmile replied:
яку саме інформацію ти дістаєш про замовлення? які поля? 
разом з ордером дістаються товари які купив користувач?

Mykhaylo12 replied:
Якщо ти про квитки, то так я поставив відповідну анотацію
 @OneToMany(fetch = FetchType.EAGER)
    private List<Ticket> tickets;

Sofasmile commented on src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java:
```diff
@@ -0,0 +1,35 @@
+    @Inject
+    private static OrderDao orderDao;
+    @Inject
+    static ShoppingCartDao shoppingCartDao;
```
```suggestion
    private ShoppingCartDao shoppingCartDao;
```

Mykhaylo12 replied:
ок

Sofasmile commented on src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java:
```diff
@@ -0,0 +1,35 @@
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private static OrderDao orderDao;
```
давай у всьому проекті видалимо static там де робимо Inject)

Mykhaylo12 replied:
добре. А можеш пояснити чому?

Sofasmile replied:
статика - зло. новий інжектор і був розроблений для того щоб позбутись використання static полів. 
https://stackoverflow.com/questions/7026507/why-are-static-variables-considered-evil

Mykhaylo12 replied:
ok

Mykhaylo12 committed:
hw27.4
-видалив статік занчення під анотацією інжектор
-видалив метод клір на рівні дао

Sofasmile commented on src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java:
```diff
@@ -5,26 +5,31 @@
         order.setUser(user);
         order.setOrderTime(LocalDateTime.now());
-        shoppingCartDao.clear(shoppingCartDao.getByUser(user));
+        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user);
```
ми можемо зробити так:
        ShoppingCart shoppingCart = shoppingCartService.getByUser(user);
і потім видалити:
private ShoppingCartDao shoppingCartDao;
